docling_core/types/experimental/document.py

```diff
@@ -0,0 +1,1106 @@
+
+                    cell_loc = ""
+                    if (
+                        col.bbox is not None
+                        and add_cell_location
+                        and add_page_index
+                        and len(self.prov) > 0
+                    ):
+                        page_w, page_h = doc.pages[self.prov[0].page_no].size.as_tuple()
+                        cell_loc = DocumentToken.get_location(
+                            bbox=col.bbox.to_bottom_left_origin(page_h).as_tuple(),
+                            page_w=page_w,
+                            page_h=page_h,
+                            xsize=xsize,
+                            ysize=ysize,
+                            page_i=self.prov[0].page_no,
+                        )
+                    elif (
+                        col.bbox is not None
+                        and add_cell_location
+                        and not add_page_index
+                        and len(self.prov) > 0
+                    ):
+                        page_w, page_h = doc.pages[self.prov[0].page_no].size.as_tuple()
+
+                        cell_loc = DocumentToken.get_location(
+                            bbox=col.bbox.to_bottom_left_origin(page_h).as_tuple(),
+                            page_w=page_w,
+                            page_h=page_h,
+                            xsize=xsize,
+                            ysize=ysize,
+                            page_i=-1,
+                        )
+
+                    cell_label = ""
+                    if add_cell_label:
+                        if col.column_header:
+                            cell_label = "<col_header>"
+                        elif col.row_header:
+                            cell_label = "<row_header>"
+                        elif col.row_section:
+                            cell_label = "<row_section>"
+                        else:
+                            cell_label = "<body>"
+
+                    body += f"<col_{j}>{cell_loc}{cell_label}{text}</col_{j}>"
+
+                body += f"</row_{i}>{new_line}"
+
+        body += f"{DocumentToken.END_TABLE.value}{new_line}"
+
+        return body
+
+
+class KeyValueItem(DocItem):
+    """KeyValueItem."""
+
+
+ContentItem = Union[TextItem, FigureItem, TableItem, KeyValueItem]
+
+
+class DocumentTrees(BaseModel):
+    """DocumentTrees."""
+
+    furniture: GroupItem = GroupItem(
+        name="_root_", dloc="#/furniture"
+    )  # List[RefItem] = []
+    body: GroupItem = GroupItem(name="_root_", dloc="#/body")  # List[RefItem] = []
+
+
+class PageItem(BaseModel):
+    """PageItem."""
+
+    # A page carries separate root items for furniture and body,
+    # only referencing items on the page
+    hash: str  # page hash
+    size: Size
+    image: Optional[ImageRef] = None
+    page_no: int
+
+
+class DoclingDocument(DocumentTrees):
+    """DoclingDocument."""
+
+    version: str = "0.0.1"  # = SemanticVersion(version="0.0.1")
```

Comment:
Some thoughts:
- This would imply manual maintenance overhead & the error-proneness inherent to that (e.g. if dev forgets to update)
- One could consider the alternative of directly using the docling-core package version, which is currently managed automatically, so above issues would not exist — however multiple package versions would correspond to the same document data model, so there would be no clear version management of the document data model itself

I therefore think we should indeed stick to the manual version here.

👉 Note though, the first version in SemVer is typically [0.1.0](https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase).

Reply:
I was hoping that we could use a specific semver type that only accepts valid version strings, but it seemed broken in [`pydantic-extra-types`](https://pypi.org/project/pydantic-extra-types/)